processor/k8sprocessor/config.go

```diff
@@ -43,59 +47,72 @@ type ExtractConfig struct {
 	// Metadata allows to extract pod metadata from a list of metadata fields.
 	// The field accepts a list of strings.
 	//
-	// Metadata fields supported right now are,
-	//   namespace, podName, deployment, cluster, node and startTime
-	//
-	// Specifying anything other than these values will result in an error.
+	// Specifying anything not-supported will result in an error.
 	// By default all of the fields are extracted and added to spans.
 	Metadata []string `mapstructure:"metadata"`
 
+	// Tags allow to specify output name used for each of the kubernetes tags
+	// The field accepts a map of string->string. It is optional and if no values
+	// are provided, defaults will be used
+	Tags map[string]string `mapstructure:"tags"`
```

Comment:
For many, the default tag names will be just fine, but some have already a taxonomy in use and the data coming through OpenTelemetry Collector should match it. I did want to keep the config compatibility with the existing format (which might be a bad assumption) and just added another section, where custom tag names can be provided. They are being held in `Tags`. 

Let's consider single field, e.g. `podName`. If someone wants to fetch it, they provide following:
```yaml
extract:
  metadata:
    - podName
```

On startup, since the config was provided, the default value will be used, i.e. `k8s.pod.name`: https://github.com/SumoLogic/opentelemetry-collector-contrib/blob/k8s-extensions/processor/k8sprocessor/kube/kube.go#L171

Whenever a record will be tagged, the name of the field will be taken from there: https://github.com/SumoLogic/opentelemetry-collector-contrib/blob/k8s-extensions/processor/k8sprocessor/kube/client.go#L198 So in this case, the pod name will be stored in `k8s.pod.name` attribute 

Now, let's say that someone wants to use `pod_name` as the tag name. Following config might be provided:
```yaml
extract:
  metadata:
    - podName
  tags:
    podName: pod_name
```

This will be handled by https://github.com/SumoLogic/opentelemetry-collector-contrib/blob/k8s-extensions/processor/k8sprocessor/options.go#L170-L171 and eventually `Tags` will include the name set by the user. So whenever a record will be tagged, the attribute key name will be `pod_name`

Reply:
There's a fair chance that I might have missed a better way to do it. On way to handle that is to explicitly provide the name each time, e.g.:
```yaml
extract:
  metadata:
    podName: pod_name
```